base_local_planner/src/latched_stop_rotate_controller.cpp

```diff
@@ -85,9 +99,14 @@ bool LatchedStopRotateController::isGoalReached(LocalPlannerUtil* planner_util,
 
   base_local_planner::LocalPlannerLimits limits = planner_util->getCurrentLimits();
 
+  const double goal_dist = base_local_planner::getGoalPositionDistance(global_pose, goal_x, goal_y);
+  const bool reached_outer_goal = goal_dist <= xy_goal_tolerance;
+  const bool reached_inner_goal = goal_dist <= inner_xy_goal_tolerance;
+  const bool bypassed_goal = isGoalBypassed(planner_util, global_pose);
+
   //check to see if we've reached the goal position
   if ((latch_xy_goal_tolerance_ && xy_tolerance_latch_) ||
-      base_local_planner::getGoalPositionDistance(global_pose, goal_x, goal_y) <= xy_goal_tolerance) {
+      (reached_outer_goal && (reached_inner_goal || bypassed_goal))) {
```

Comment:
OOS / Q: do you know why this logic is duplicated and not just `isPositionReached` is used, apart from the debug statement? (wouldn't change it in this PR though, as OOS)

Reply:
hm that's a good point, not sure why the code is duplicated